### ISSUE TITLE
test/testutil/testutil_init.c does not compile with enable-trace

### ISSUE THREAD

**Issue by DDvO** [openssl/openssl#10731]
When configured with `enable_trace` I get 
```
gcc  -Iinclude -Iapps/include -I.  -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O0 -g -DDEBUG_UNUSED -DPEDANTIC -pedantic -Wno-long-long -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wswitch -Wsign-compare -Wshadow -Wformat -Wtype-limits -Wundef -Werror -Wmissing-prototypes -Wstrict-prototypes -std=gnu90 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib/engines-3\"" -DMODULESDIR="\"/usr/local/lib/ossl-modules\""   -MMD -MF test/testutil/libtestutil-lib-testutil_init.d.tmp -MT test/testutil/libtestutil-lib-testutil_init.o -c -o test/testutil/libtestutil-lib-testutil_init.o test/testutil/testutil_init.c
test/testutil/testutil_init.c: In function 'internal_trace_cb':
test/testutil/testutil_init.c:39:35: error: 'PREFIX_CTRL_SET_PREFIX' undeclared (first use in this function); did you mean 'BIO_CTRL_SET_PREFIX'?
         BIO_ctrl(trace_data->bio, PREFIX_CTRL_SET_PREFIX,
                                   ^~~~~~~~~~~~~~~~~~~~~~
                                   BIO_CTRL_SET_PREFIX
test/testutil/testutil_init.c:39:35: note: each undeclared identifier is reported only once for each function it appears in
test/testutil/testutil_init.c: In function 'setup_trace_category':
test/testutil/testutil_init.c:79:32: error: implicit declaration of function 'apps_bf_prefix'; did you mean 'BIO_f_prefix'? [-Werror=implicit-function-declaration]
     channel = BIO_push(BIO_new(apps_bf_prefix()),
                                ^~~~~~~~~~~~~~
                                BIO_f_prefix
test/testutil/testutil_init.c:79:32: error: passing argument 1 of 'BIO_new' makes pointer from integer without a cast [-Werror=int-conversion]
     channel = BIO_push(BIO_new(apps_bf_prefix()),
                                ^~~~~~~~~~~~~~~~
In file included from include/openssl/trace.h:15,
                 from test/testutil/testutil_init.c:12:
include/openssl/bio.h:587:6: note: expected 'const BIO_METHOD *' {aka 'const struct bio_method_st *'} but argument is of type 'int'
 BIO *BIO_new(const BIO_METHOD *type);
      ^~~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:20369: test/testutil/libtestutil-lib-testutil_init.o] Error 1
```

**Comment by DDvO** [comment]
Thanks @mspncp for your immediate reaction.

**Comment by DDvO** [comment]
Since it is not the first time that a build issue goes unnoticed for a while when `OPENSSL_NO_TRACE` is not defined,
would be really good if the configuration option `enable_trace` was part of the CI builds.

**Comment by mspncp** [comment]
> Thanks @mspncp for your immediate reaction.

You're welcome, always a pleasure. See #10732.

**Comment by mspncp** [comment]
> Since it is not the first time that a build issue when `OPENSSL_NO_TRACE` is not defined goes unnoticed for a while,
> would be really good if the configuration option `enable_trace` was part of the CI builds.

Can you raise a separate issue for this please, so it won't be forgotten?

**Comment by DDvO** [comment]
> Can you raise a separate issue for this please, so it won't be forgotten?

I did already on Oct 22nd: #10237.
I've just added a cross-reference there.
